components/Header/__tests__/SearchBar.test.tsx

```diff
@@ -1,62 +1,63 @@
 import { render, screen } from '@testing-library/react';
 import { SearchBar } from '../searchBar';
 import userEvent from '@testing-library/user-event';
-import { GlobalContext } from '@/context/GlobalStore';
+import * as GlobalContext from '@/context/GlobalStore';
 
 const user = userEvent.setup();
+const mockCategories = [
+  { ID: '1', name: 'Hello', subCategories: [] },
+  { ID: '2', name: 'World', subCategories: [] },
+];
+
+describe('Search Bar Component', () => {
+  beforeEach(() => {
+    jest.spyOn(GlobalContext, 'useGlobalStore').mockReturnValue({
+      state: { brands: [], categories: mockCategories, cart: [] },
+      dispatch: jest.fn,
+    });
+  });
+
+  it('matches snapshot', () => {
+    const { asFragment } = render(<SearchBar />);
 
-const MockComponent = () => {
-  const mockCategories = [
-    { ID: '1', name: 'Hello', subCategories: [] },
-    { ID: '2', name: 'World', subCategories: [] },
-  ];
-
-  return (
-    <GlobalContext.Provider
-      value={{
-        state: { categories: mockCategories, brands: [], cart: [] },
-        dispatch: () => {},
-      }}>
-      <SearchBar />
-    </GlobalContext.Provider>
-  );
-};
-
-// describe('Search Bar Component', () => {
-it('renders a category button', () => {
-  render(<SearchBar />);
-
-  const categoryButton = screen.getByRole('button', {
-    name: /all categories/i,
+    expect(asFragment()).toMatchSnapshot();
   });
 
-  expect(categoryButton).toBeInTheDocument();
-});
+  it('renders a category button', () => {
+    const { asFragment } = render(<SearchBar />);
 
-it('renders a category menu when category button is clicked', async () => {
-  render(<MockComponent />);
+    const categoryButton = screen.getByRole('button', {
+      name: /all categories/i,
+    });
 
-  const categoryButton = screen.getByRole('button', {
-    name: /all categories/i,
+    expect(categoryButton).toBeInTheDocument();
+    expect(asFragment()).toMatchSnapshot();
```

Comment:
You have already created a snapshot for this component above, if the component renders the same here then it is redundant, remove it if that's the case